docs/modules/ROOT/pages/how-tos/opsgenie.adoc

```diff
@@ -0,0 +1,263 @@
+= Integrating Alertmanager with OpsGenie
+
+This how-to assumes that you're already using OpsGenie.
+It additionally assumes that you're familiar with managing schedules and escalations.
+
+== Prerequisites
+
+In order to setup Alertmanager integration with OpsGenie, you'll need sufficient permissions in OpsGenie to setup integrations and heartbeats.
+If you're using teams in OpsGenie, request role "admin" in your team to be able to follow along.
+
+== Configuring the integration in OpsGenie
+
+In OpsGenie, navigate to "Teams > Your Team > Integrations" and enable the Prometheus and REST API integrations.
+You can use the same integrations to receive alerts from multiple Alertmanagers.
+Save the API keys which are generated when enabling the integrations in Vault, we'll reference them from the Commodore hierarchy.
+In the REST API integration enable "Read Access" and "Create and Update Access."
+The REST API integration is required for heartbeat alerts, such as the "Watchdog" alert.
+
+Configure a heartbeat in "Teams > Your Team > Heartbeats" to receive `Watchdog` alerts as heartbeats in OpsGenie.
+Create a new heartbeat, the name can be whatever you'd like.
+We suggest using the Commodore `<cluster-id>` of the cluster you're integrating with OpsGenie as the heartbeat nam e.
+Set the heartbeat interval to two minutes.
+Optionally, you can add the cluster's display name (or any other descriptive name for the cluster) in the heartbeat's description field.
+
+== Configuring Alertmanager to send alerts to OpsGenie
+
+To configure Alertmanager to send alerts to OpsGenie we need to configure an Alertmanager receiver using Alertmanager's `opsgenie` integration.
+First, we need to configure the `opsgenie_api_key` in the `global` section of the Alertmanager config.
+For the global API key you'll want to reference the Vault secret holding the API key for the Prometheus integration.
+
+[source,yaml]
+----
+openshift4_monitoring:
+  alertManagerConfig:
+    global:
+      opsgenie_api_key: ?{vaultkv:${cluster:tenant}/${cluster:name}/opsgenie/api-key}
+----
+
+Next, we need to configure the OpsGenie receiver.
+First off, we want to set our team as the responder for all the alerts received from Alertmanager.
+We recommend that you use the team's UUID for configuring the responders, as the UUID remains stable even if the team name is changed.
+We also configure the OpsGenie receiver as the default receiver.
+
+[source,yaml]
+----
+openshift4_monitoring:
+  alertManagerConfig:
+    receivers:
+      - name: opsgenie
+        opsgenie_configs:
+          - responders:
+              - id: <team-uuid>
+                type: team
+    route:
+      receiver: opsgenie
+----
+
+Additionally, we configure a number of things to make use of Project Syn and Openshift4 conventions in the alerts which are created in OpsGenie.
+
+In order for the snippets discussed below to work correctly, we need to group alerts by `alertname`, `namespace`, and `severity` at least.
```

Comment:
This is subject to discuss and teams are free to handle this differently. Can you clarify that this is a common ground to start with?

Reply:
That we need to discuss this internally and that our teams are free to do something different is out of scope for this how-to/PR. In general, a how-to should always be taken as a starting point for building a configuration that meets the needs of the person/organization who consumes it.

Grouping alerts by *at least* `alertname`, `namespace` and `severity` is mandatory in the scope of this how-to, as otherwise the `.GroupLabels` references in the configuration presented by the how-to will not work correctly.